centraldogma/src/test/java/com/linecorp/decaton/centraldogma/CentralDogmaPropertySupplierTest.java

```diff
@@ -133,4 +145,111 @@ public void testGetPropertyAbsentName() {
         PropertyDefinition<Object> missingProperty = PropertyDefinition.define("absent.value", Long.class);
         assertFalse(supplier.getProperty(missingProperty).isPresent());
     }
+
+    @Test
+    public void testRegisterWithDefaultSettings() {
+        when(centralDogma.normalizeRevision(PROJECT_NAME, REPOSITORY_NAME, Revision.HEAD)).thenReturn(
+                CompletableFuture.completedFuture(Revision.HEAD)
+        );
+        when(centralDogma.listFiles(PROJECT_NAME, REPOSITORY_NAME, Revision.HEAD, FILENAME)).thenReturn(
+                CompletableFuture.completedFuture(Collections.emptyMap())
+        );
+        when(centralDogma.push(
+                eq(PROJECT_NAME),
+                eq(REPOSITORY_NAME),
+                eq(Revision.HEAD),
+                any(String.class),
+                eq(Change.ofJsonUpsert(FILENAME, defaultPropertiesAsJsonNode())))
+        ).thenReturn(
+                CompletableFuture.completedFuture(
+                        new PushResult(Revision.HEAD, 1)
+                )
+        );
+
+        CentralDogmaPropertySupplier.register(centralDogma, PROJECT_NAME, REPOSITORY_NAME, FILENAME);
+        verify(centralDogma, times(1)).push(
+                eq(PROJECT_NAME),
+                eq(REPOSITORY_NAME),
+                eq(Revision.HEAD),
+                any(String.class),
+                eq(Change.ofJsonUpsert(FILENAME, defaultPropertiesAsJsonNode()))
+        );
+
+    }
+
+    @Test
+    public void testRegisterWithCustomizedSettings() {
+        final int settingForPartitionConcurrency = 188;
+        final int settingForMaxPendingRecords = 121212;
+        final int whenCentralDogmaPushed = 111111;
+
+        List<Property<?>> properties = ProcessorProperties
```

Comment:
I don't see why we need full list of proeprties here, not only custom ones but also these with defaults?
Since the spec of newly added `register` API is such that "Creates a node with default properties, but you can specify some prop's initial value optionally", the realistic usage will look like:

```java
CentralDogmaPropertySupplier.register(..., StaticPropertySupplier.of(
    Property.ofStatic(...)
));
```

so what we should test here is passing just partial list of properties that we want to customize? such as:


```java
    @Test
    public void testRegisterWithCustomizedSettings() {
        final int settingForPartitionConcurrency = 188;
        final int settingForMaxPendingRecords = 121212;
        final int whenCentralDogmaPushed = 111111;

        final PropertySupplier supplier = StaticPropertySupplier.of(
                Property.ofStatic(ProcessorProperties.CONFIG_PARTITION_CONCURRENCY,
                                  settingForPartitionConcurrency),
                Property.ofStatic(ProcessorProperties.CONFIG_MAX_PENDING_RECORDS,
                                  settingForMaxPendingRecords)
        );
        final List<Property<?>> listProperties = ProcessorProperties
                .defaultProperties()
                .stream()
                .map(defaultProperty -> {
                    Optional<? extends Property<?>> prop = supplier.getProperty(defaultProperty.definition());
                    if (prop.isPresent()) {
                        return prop.get();
                    } else {
                        return defaultProperty;
                    }
                }).collect(Collectors.toList());

...
```

Reply:
I agree with you. I don't need to provide all of the properties if I don't change the value of it.
However, on the other hand I have to build a full list of properties for verifying mock.

Changed: providing only customized properties
e6ae753